java/parser.js

```diff
@@ -1,171 +1,206 @@
+        actualOptions.forEach(option => {
+            if (!supported.includes(option)) throw new Error(`Emitter need update to support check settings option: ${option}`);
+        })
+    }
+
 }
 
 // General
 
 function java(chunks, ...values) {
-    const commands = []
-    let code = ''
+    const commands = [];
+    let code = '';
     values.forEach((value, index) => {
         if (typeof value === 'function' && !value.isRef) {
-            code += chunks[index]
-            commands.push(code, value)
-            code = ''
+            code += chunks[index];
+            commands.push(code, value);
+            code = '';
         } else {
-            code += chunks[index] + serialize(value)
+            code += chunks[index] + serialize(value);
         }
     })
-    code += chunks[chunks.length - 1]
-    commands.push(code)
-    return commands
+    code += chunks[chunks.length - 1];
+    commands.push(code);
+    return commands;
 }
 
 function serialize(value) {
-    let stringified = ''
+    let stringified = '';
     if (value && value.isRef) {
-        stringified = value.ref()
+        stringified = value.ref();
     } else if (value === null) {
-        throw Error(`Null shouldn't be passed to the java code. \n ${value}`)
+        throw Error(`Null shouldn't be passed to the java code. \n ${value}`);
     } else if (typeof value === 'object') {
-        stringified = parseObject(value)
+        stringified = parseObject(value);
     } else if (typeof value === 'function') {
-        stringified = value.toString()
+        stringified = value.toString();
     } else if (typeof value === 'undefined') {
-        throw Error(`Undefined shouldn't be passed to the java code. \n ${value}`)
+        throw Error(`Undefined shouldn't be passed to the java code. \n ${value}`);
     } else {
-        stringified = JSON.stringify(value)
+        stringified = JSON.stringify(value);
     }
-    return stringified
+    return stringified;
 }
 
 function parseObject(object) {
     if (object.selector) {
-        return selectors[object.type](JSON.stringify(object.selector))
+        return selectors[object.type](JSON.stringify(object.selector));
     } else if (object.type) {
-        const typeBuilder = types[object.type]
+        const typeBuilder = types[object.type];
         if (typeBuilder) {
             if (typeBuilder.isGeneric) {
-                return typeBuilder.constructor(object.value, object.generic)
+                return typeBuilder.constructor(object.value, object.generic);
             } else {
-                return typeBuilder.constructor(object.value)
+                return typeBuilder.constructor(object.value);
             }
-        } else throw new Error(`Constructor wasn't implemented for the type: ${object.type}`)
-    } else return JSON.stringify(object)
+        } else throw new Error(`Constructor wasn't implemented for the type: ${object.type}`);
+    } else return JSON.stringify(object);
```

Comment:
Codacy has a fix for the issue: [Expected { after 'else'.](https://app.codacy.com/gh/applitools/sdk.coverage.tests/pullRequest?prid=8528488)
```suggestion
    } else {return JSON.stringify(object);}
```